### ISSUE TITLE
[Bug]: Linter does not pick up accurate errors when using operators together without paranthesis

### ISSUE THREAD

**Issue by ramsaptami** [appsmithorg/appsmith#15339]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Linter does not pick up accurate errors when using operators together without paranthesis

### Steps To Reproduce

![image](https://user-images.githubusercontent.com/79509062/180003191-e5a3d4bb-b22d-45c8-8a7e-d4b33c8881c9.png)

### Version

Cloud

**Comment by bharath31** [comment]
@ramsaptami this would be a syntax error. both `??` and `||` operators have the same [precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#table) and should be used with parentheses in an expression

**Comment by ramsaptami** [comment]
That's correct @bharath31 and the error we see does not say anything about syntax which needs to be improved here.
